sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -243,6 +243,17 @@ object ResolveDefaultColumns extends QueryErrorsBase
     getDefaultValueExprOrNullLit(field, useNullAsDefault)
   }
 
+  def generateSQL(defaultValue: DefaultValue): String = {
+    if (defaultValue.getSql != null) {
+      defaultValue.getSql
+    } else {
+      ExpressionConverter.toV1(defaultValue.getExpression) match {
+        case Some(e) if !e.isInstanceOf[NonSQLExpression] => e.sql
```

Comment:
This seems wrong as `NonSQLExpression` can be nested.

If we don't allow connectors to return v2 expression as the default value for now, maybe we can fail first and support it later?

Reply:
OK we have a test for it. Maybe we should refactor `ResolveDefaultColumns.analyze` and make it accept `Expression` directly.

Reply:
I was thinking about this too. I updated `ResolveDefaultColumns` to accept an expression.